### PR TITLE
Add caddy container to handle https

### PR DESCRIPTION
**Pull request by treyenelson**

### CODE REVIEW

**Review by ferdnyc — COMMENTED** [review]
Just as a general question, doesn't this break the ability to run a server on a host that _already_ has SSL certificates? (Generated either with Let's Encrypt, or some other method.) I'm not familiar with Caddy specifically, but just from reading over these changes it seems as though the ability to _define_ the SSL certificates to be used is entirely replaced with Caddy autogeneration. Which sounds like it's not always what everyone would want, especially in production.

File: README.md [hunk 1]
```diff
@@ -176,26 +175,20 @@ newgrp docker
...
+# Start the server
+docker compose up --build -d
 
 # Create the database and initialize the static files
 docker compose exec django python manage.py migrate
 docker compose exec django python manage.py collectstatic
 
 # EITHER initialize the data
-docker compose exec python manage.py createsuperuser
+docker compose exec django python manage.py createsuperuser
```
Comment (ferdnyc) [inline README.md:191]: I feel like corrections like this should be submitted in a separate commit, if not a separate PR. This particular change, for example, is totally unrelated to Caddy, right? It's just a documentation correction?

**Comment by pennomi** [comment]
That's likely, though it's fairly trivial to override this if you have a specific need. Before merging I will make sure there's a documented method for people who want to bring their own certs.

Another issue that I'm working on here is ensuring that the server responds to traffic on all configured domains rather than just the one, since this project allows multiple comics to be hosted at once, and has redirect "alias" domains.

File: README.md [hunk 2]
```diff
@@ -34,16 +33,16 @@ You now have a server running. Unfortunately it doesn't do much - we need to ini
 
 ## Backing Up and Restoring Data
 
-To back up your database, run `docker compose exec comics python manage.py backup dump <your filename here>.zip`.
+To back up your database, run `docker compose exec django python manage.py backup dump <your filename here>.zip`.
 
 To restore your database from a backup zip:
  - Ensure you have an empty database to start. It should be migrated.
- - Then load the backup zip `docker-compose exec comics python manage.py backup load <your filename here>.zip`.
+ - Then load the backup zip `docker-compose exec django python manage.py backup load <your filename here>.zip`.
```
Comment (ferdnyc) [inline README.md:40]: Same.

File: README.md [hunk 3]
```diff
@@ -15,15 +15,14 @@ If you're unfamiliar with Docker, this might be a little tricky for you. It's wo
...
-- Run `docker compose up -d`
+- Run `docker compose up --build -d`
 
 You now have a server running. Unfortunately it doesn't do much - we need to initialize the data or restore from a backup.
 
 
 ## Initialize Data From Scratch
 
-- Enter a shell on the docker django machine `docker compose exec comics bash`
+- Enter a shell on the docker django machine `docker compose exec django bash`
```
Comment (ferdnyc) [inline README.md:25]: Here too.
